tube/etl/indexers/injection/parser.py

```diff
@@ -123,10 +123,22 @@ def update_level(self):
             new_assigned = set([])
             for collector in just_assigned:
                 for child in collector.children:
-                    if len(child.children) == 0 or child in assigned_levels:
+                    # TODO: PXP-5067 investigate int-string comparison in node lvls
```

Comment:
I don't understand your comment here. Why do we need to change the logic here? what is the bug?